hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/HoodieIncrSource.java

```diff
@@ -231,7 +243,15 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkpt
     // Remove Hoodie meta columns except partition path from input source
     String[] colsToDrop = shouldDropMetaFields ? HoodieRecord.HOODIE_META_COLUMNS.stream().toArray(String[]::new) :
         HoodieRecord.HOODIE_META_COLUMNS.stream().filter(x -> !x.equals(HoodieRecord.PARTITION_PATH_METADATA_FIELD)).toArray(String[]::new);
-    final Dataset<Row> src = source.drop(colsToDrop);
+    Dataset<Row> src = source.drop(colsToDrop);
+    if (getLatestSourceProfile().isPresent()) {
+      src = coalesceOrRepartition(src, getLatestSourceProfile().get().getSourcePartitions());
+    }
```

Comment:
Could `getLatestSourceProfile().map().orElse()` be used instead of reassigning the variable?